Feature/get full pub.dev score

nohli opened the pull request:
This PR updates the `analyzer_plugin` dependency, resulting in a full `pub.dev` score.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/invertase/dart_custom_lint?pullRequest=27) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/invertase/dart_custom_lint?pullRequest=27) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/invertase/dart_custom_lint?pullRequest=27) it.</sub>

nohli commented:
@rrousselGit I've updated `analyzer` and `analyzer_plugin` locally and fixed all lints.
[The diff is here.](https://github.com/invertase/dart_custom_lint/compare/main...nohli:dart_custom_lint:feature/get-full-pub.dev-score)

Just two things:
1. I'm not sure about the line `unawaited(driver?.dispose2());` - should the Future `dispose2()` be awaited or not?
2. The `custom_lint` dependency in `custom_lint_builder`'s `pubspec.yaml` would need to be updated to a version depending on `  analyzer_plugin: ^0.11.1`.

Would this be good for a new PR, or are there more changes than fixing lints?
